### DISCUSSION TITLE
Is it too small to configure the constant maxConcurrentAsyncSend to 50000

### DISCUSSION THREAD

**Discussion by FourSpaces** [minio/minio#19292] [Q&A] [answered]
I encountered an error while using the DeleteMultipleObjects interface to delete a large number of objects.
```
API: SYSTEM
Time: 08:33:16 UTC 03/13/2024
DeploymentID: 79016fd5-3621-43ef-8c3a-77f21b28eed7
RequestID: 
RemoteHost: 
UserAgent: 
Error: concurrent target notifications exceeded 50000, notification endpoint is too slow to accept events on incoming requests (*errors.errorString)

Backtrace: 
0: internal/logger/logonce.go:118:logger.(*logOnceType).logOnceIf()
1: internal/logger/logonce.go:149:logger.LogOnceIf()
2: internal/event/targetlist.go:316:event.(*TargetList).sendAsync()
3: internal/event/targetlist.go:268:event.(*TargetList).Send()
4: cmd/event-notification.go:156:cmd.(*EventNotifier).Send()
5: cmd/event-notification.go:264:cmd.sendEvent()
6: cmd/bucket-handlers.go:699:cmd.objectAPIHandlers.DeleteMultipleObjectsHandler()
7: net/http/server.go:2136:http.HandlerFunc.ServeHTTP()
```

I have looked at some of the code, and the queue length of TargetList is equal to the constant maxConcurrentAsyncSend. Is it too small to configure the constant maxConcurrentAsyncSend to 50000?

Why not set more? What would be the problem if I changed to a larger queue here?

**Comment by harshavardhana** [comment, marked as answer]
Your notification target is unable to accept the requests causing a large build-up on the MinIO side, that's why we are showing this message. 

>  notification endpoint is too slow to accept events for incoming request rate
